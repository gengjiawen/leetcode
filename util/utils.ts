export function underline(s: string) {
  return s
    .split(' ')
    .map((a) => a.toLowerCase())
    .join('_')
    .replaceAll('?', '')
    .replaceAll('-', '_')
}

export function extract_rust_solution(str: string) {
  const regex = /pub fn .*?}/gms
  if (!str.startsWith('impl Solution')) {
    return str
  }
  const r = str.match(regex)
  if (r) {
    return r[0]
  }
  return str
}
