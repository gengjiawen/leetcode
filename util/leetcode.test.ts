import { extract_rust_solution, underline } from './utils'

test('test underline filename', () => {
  const a = underline('minimum-swaps-to-group-all-1s-together')
  expect(a).toBe('minimum_swaps_to_group_all_1s_together')
})

test('extract rust solutions', () => {
  const str = `impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {

    }
}`
  const r = extract_rust_solution(str)
  expect(r).toMatchSnapshot()
})

test('reference test', () => {
  function append_str(o: any) {
    o.str += '__append'
  }

  const o = { str: 'hi' }
  append_str(o)
  console.log(o)
})
