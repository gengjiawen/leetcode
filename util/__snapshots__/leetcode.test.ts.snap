// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract rust solutions 1`] = `
"pub fn climb_stairs(n: i32) -> i32 {

    }"
`;
